src/clr-angular/data/datagrid/providers/column-order-model.service.mock.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ColumnOrderModelService } from './column-order-model.service';
+
+export class MockColumnOrderModelService {
```

Comment:
`export class MockColumnOrderModelService implements ColumnOrderModelService`
It will ensure we keep the mock in sync with the actual service in case of future updates.

Reply:
`ColumnOrderModelService` is a class though. We could only implement interfaces though, no? Extending a class wouldn't serve the purpose you are describing here, I think.

Reply:
Yes, you can use any class or or abstract class as an interface if you want.
So the point here really is to `implement`, not `extend`. It just ensures that if we add a new method to the `ColumnOrderModelService`, the mock will fail to compile because it doesn't implement all the methods of the original service. Failing at compile time is way better than running 15 minutes of test to see it fail with an obscure error message.